### PR TITLE
fix spurious test failure and add improved debug

### PR DESCRIPTION
**Pull request by garlick**
Zeroing `pmix_value_t` storage before use seems to both shut up valgrind and resolve issue #69.  As pointed out by @grondo, the supposed fix may just be eliminating spurious valgrind warnings.  Why the issue goes away is unknown, but maybe there's some other effect going on in the pmix server code that I'm missing.  I guess we can reopen if it recurs.

### CODE REVIEW

File: src/shell/plugins/fence.c [hunk 1]
```diff
@@ -102,37 +102,24 @@ static void exchange_exit_cb (struct exchange *xcg, void *arg)
...
-     * osu benchmarks.  We don't know what to do with it, but let's
-     * tone down the warning to debug level to avoid alarming users.
-     */
-    else if (!strcmp (info->key, "pmix.loc.col.st")) {
-        shell_debug ("ignoring experimental %s fence attr: %s",
-                     required ? "required" : "optional",
-                     info->key);
-        goto done;
+    else if ((info->flags & PMIX_INFO_REQD)) {
+        shell_warn ("unknown required fence attr: %s", info->key);
```
Comment (grondo) [inline src/shell/plugins/fence.c:116]: There's two error messages that are added here at `shell_warn()`, but the commit message says this commit should be replacing `shell_warn()` calls with `shell_debug()`. Not sure if the commit message just needs an update or if these changes were meant to be in a separate commit? (Or perhaps I'm misreading the commit message)

**Review by grondo — APPROVED** [review]
LGTM! One question about a commit message inline.

File: src/shell/plugins/fence.c (hunk 1 above)
Comment (garlick) [reply src/shell/plugins/fence.c:116]: The commit does do what it says, but it's not so clear from the diff because the change allowed me to simplify the function.  There are now three legs to the if clause:  1) a known attribute (includes an error logged at warn level if it's the wrong type), 2) attributes that are required but unknown (logged at warn level), and 3) attributes that are not required and unknown (logged at debug level).

Does that make sense?

Comment (grondo) [reply src/shell/plugins/fence.c:116]: Works for me!

**Comment by garlick** [comment]
Thanks, setting MWP!
